api/src/wfl/module/covid.clj

```diff
@@ -27,6 +28,37 @@
 (defn start-covid-workload!
   [])
 
+(defn ^:private snapshot-imported?
+  "Check if snapshot SNAPSHOT_ID in TDR has been imported to WORKSPACE
+  as SNAPSHOT_REFERENCE_ID and can be successfully fetched."
+  [{:keys [snapshot_id snapshot_reference_id] :as _source_details}
+   {:keys [workspace] :as _executor}]
+  (letfn [(fetch [] (rawls/get-snapshot-reference workspace
+                                                  snapshot_reference_id))]
+    (some? (and snapshot_id snapshot_reference_id (util/do-or-nil (fetch))))))
+
+(defn ^:private import-snapshot!
+  "Import snapshot SNAPSHOT_ID from TDR to WORKSPACE.
+  Use transaction TX to update DETAILS table with resulting reference id.
+  Throw if import fails."
+  [tx
+   {:keys [workload_id] :as _workload}
+   {:keys [details] :as _source}
+   {:keys [details_id snapshot_id] :as source_details}
+   {:keys [workspace] :as _executor}]
+  (letfn [(create! [] (rawls/create-snapshot-reference workspace snapshot_id))]
```

Comment:
I punted naming the snapshot reference to `rawls`, which will generate the name using the TDR snapshot name as its base.

Reply:
The snapshot already has a name (it must be named when created). If the executor is given the snapshot object (i.e. that returned by `datarepo/get-snapshot` which contains the name) you can just use that.

Reply:
Yes, that's what I'm doing.  See change to `rawls/create-snapshot-reference` which calls `datarepo/get-snapshot` to get the name if none is specified.